dns_entry_checker/test/test_dns_entry_checker.py

```diff
@@ -0,0 +1,193 @@
+import dns_entry_checker
+import unittest
+from unittest import mock
+from unittest.mock import MagicMock, patch, NonCallableMock, call
+from parameterized import parameterized
+from dns_entry_checker import (
+    create_client,
+    ssh_command,
+    find_ip_dns_pair,
+    check_ip_dns_mismatch,
+    check_missing_ips,
+)
+
+
+class DNSEntryCheckerTests(unittest.TestCase):
+    @patch("dns_entry_checker.paramiko")
+    def test_create_client(self, mock_paramiko):
+        host = "test.aquilon.gridpp.rl.ac.uk"
+        user = "test.abc12345"
+        password = "test.A4gdfGs"
+
+        mock_client = MagicMock()
+        mock_paramiko.client.SSHClient.return_value = mock_client
+        mock_paramiko.client.set_missing_host_key_policy = MagicMock()
+        mock_paramiko.AutoAddPolicy = NonCallableMock()
+        mock_paramiko.connect = MagicMock()
+
+        client = create_client(host, user, password)
+
+        client.connect.assert_called_once_with(
+            hostname=host, username=user, password=password, timeout=60
+        )
+
+        self.assertEqual(client, mock_client)
+
+    def test_ssh_command(self):
+        mock_client = MagicMock()
+        command = "test aq show_host --hostname localhost"
+
+        mock_stdout = MagicMock()
+        mock_client.exec_command.return_value = MagicMock(), mock_stdout, MagicMock()
+        mock_stdout.channel.recv_exit_status.return_value = NonCallableMock()
+        mock_stdout.readlines.return_value = NonCallableMock()
+
+        command_output = ssh_command(mock_client, command)
+
+        mock_client.exec_command.assert_called_once_with(
+            command, get_pty=True, timeout=60
+        )
+        mock_stdout.channel.recv_exit_status.assert_called_once()
+        mock_stdout.readlines.assert_called_once()
+
+        self.assertEqual(command_output, mock_stdout.readlines.return_value)
+
+    def test_pair_ip_and_dns(self):
```

Comment:
rename this test method so its more descriptive
```suggestion
    def test_find_ip_dns_pair_valid(self):
```